src/PNE/pne_ecrire_jeu_de_donnees_mps.c

```diff
@@ -149,35 +164,47 @@ N : free ??
 fprintf(Flot," N  OBJECTIF\n");
 /* Ecriture de toutes les contraintes */
 for ( Cnt = 0 ; Cnt < NombreDeContraintes ; Cnt++ ) {
+  int NomDeContrainteParDefaut = NomsDesContraintes == NULL || NomsDesContraintes[Cnt] == NULL;
+  MettreAjourNom(NomDeContrainteParDefaut, NomContrainte,NomsDesContraintes[Cnt], Cnt, 'R' );
   if ( Sens[Cnt] == '=' ) {
-    fprintf(Flot," E  R%07d\n",Cnt);
+    fprintf(Flot, " E  %s\n", NomContrainte);
   }
   else if (  Sens[Cnt] == '<' ) {
-    fprintf(Flot," L  R%07d\n",Cnt);
+    fprintf(Flot, " L  %s\n", NomContrainte);
   }
   else if (  Sens[Cnt] == '>' ) {
-    fprintf(Flot," G  R%07d\n",Cnt);
+    fprintf(Flot, " G  %s\n", NomContrainte);
   }
   else {
     fprintf(Flot,"PNE_EcrireJeuDeDonneesMPS : le sens de la contrainte %c ne fait pas partie des sens reconnus\n",
             Sens[Cnt]);
     exit(0);
   } 
+  
 }
 
 /* COLUMNS */
 fprintf(Flot,"COLUMNS\n");
 for ( Var = 0 ; Var < NombreDeVariables ; Var++ ) {
-  if ( CoutLineaire[Var] != 0.0 ) {
-    sprintf(Nombre,"%-.10lf",CoutLineaire[Var]);
+  int NomDeVariableParDefaut = NomsDesVariables == NULL || NomsDesVariables[Var] == NULL;
+  
+  MettreAjourNom(NomDeVariableParDefaut, NomVariable, NomsDesVariables[Var], Var, 'C');
+  if (CoutLineaire[Var] != 0.0)
+  {
+    sprintf(Nombre, "%-.10lf", CoutLineaire[Var]);
     /*Nombre[12] = '\0';*/ /* <- On prefere ne pas ajouter de troncature */
-    fprintf(Flot,"    C%07d  OBJECTIF  %s\n",Var,Nombre);
+    fprintf(Flot, "    %s  OBJECTIF  %s\n", NomVariable, Nombre);
   }
   il = Cdeb[Var];
-  while ( il >= 0 ) {
-    sprintf(Nombre,"%-.10lf",CoefficientsDeLaMatriceDesContraintes[il]);
+  while (il >= 0)
+  {
+    int NomDeContrainteParDefaut = NomsDesContraintes == NULL || NomsDesContraintes[Cnt] == NULL;
+    MettreAjourNom(NomDeContrainteParDefaut, NomContrainte, NomsDesContraintes[NumeroDeContrainte[il]], NumeroDeContrainte[il], 'R');
```

Comment:
```suggestion
    Cnt = NumeroDeContrainte[il];
    int NomDeContrainteParDefaut = NomsDesContraintes == NULL || NomsDesContraintes[Cnt] == NULL;
    MettreAjourNom(NomDeContrainteParDefaut, NomContrainte, NomsDesContraintes[Cnt], NumeroDeContrainte[il], 'R');
```
I think `Cnt` has a random value from a previous loop before the proposed change, to be checked.